### DISCUSSION TITLE
How to run nx affected base from the first commit?

### DISCUSSION THREAD

**Discussion by tomavic** [nrwl/nx#16327] [Show and Tell]
how to run nx affected base from the first commit? 

This is required as I have added CI scripts to my current project. and whenever I run the affected command, it is showing that 

Base SHA
the commit before the latest commit

Head SHA
Latest commit

**Comment by marko-hologram** [comment]
You can manually specify what commits/branches NX compares.

https://nx.dev/concepts/affected#ci

You can specify `--head` and `--base` CLI flags and I believe those can use git commit SHA identifiers. If you run `nx affected` on "main" branch for example, I think it will just default to comparing latest commit and one before it.
